src/deep_learning_container.py

```diff
@@ -83,27 +89,76 @@ def _retrieve_instance_region():
     return region
 
 
+def _retrieve_device():
+    return (
+        "gpu"
+        if os.path.isdir("/usr/local/cuda")
+        else "eia"
+        if os.path.isdir("/opt/ei_tools")
+        else "neuron"
+        if os.path.exists("/usr/local/bin/tensorflow_model_server_neuron")
+        else "cpu"
+    )
+
+
+def _retrieve_cuda():
+    cuda_version = ""
+    try:
+        cuda_path = os.path.basename(os.readlink("/usr/local/cuda"))
+        cuda_version_search = re.search(r"\d+\.\d+", cuda_path)
+        cuda_version = "" if not cuda_version_search else cuda_version_search.group()
+    except Exception as e:
+        logging.error(f"Failed to get cuda path: {e}")
+    return cuda_version
+
+
+def _retrieve_os():
+    version = ""
+    name = ""
+    with open("/etc/os-release", "r") as f:
+        for line in f.readlines():
+            if re.match(r"^ID=\w+$", line):
+                name = re.search(r"^ID=(\w+)$", line).group(1)
+            if re.match(r'^VERSION_ID="\d+\.\d+"$', line):
+                version = re.search(r'^VERSION_ID="(\d+\.\d+)"$', line).group(1)
```

Comment:
What about Amazon Linux? Shall we handle that case as well?

Reply:
Let's cross that bridge when we come to it.

Right now, none of the rest of our code is capable of handling Amazon Linux, either.